x/evm/keeper/statedb.go

```diff
@@ -572,20 +572,10 @@ func (k *Keeper) AddLog(log *ethtypes.Log) {
 	key := log.TxHash
 
 	if len(k.ctx.TxBytes()) > 0 {
-		tx := &ethtypes.Transaction{}
-		if err := tx.UnmarshalBinary(k.ctx.TxBytes()); err != nil {
-			k.Logger(k.ctx).Error(
-				"ethereum tx unmarshaling failed",
-				"error", err,
-			)
-			return
-		}
```

Comment:
this should be the same logic as the encoding config, can you check why is it failing? is it because the `TxBytes` are different or because they are always empty?

Reply:
I think the `TxBytes` is serialized form of [`sdk.Tx`](https://github.com/cosmos/cosmos-sdk/blob/e17be874bb0e3a246b74752e9b8894855cab9b03/types/tx/tx.pb.go#L32), rather than the `MsgEthereumTx`.